v0/destinations/af/deleteUsers.js

```diff
@@ -0,0 +1,159 @@
+const { generateUUID } = require("../../util");
+const ErrorBuilder = require("../../util/error");
+
+/**
+ * This function is making the ultimate call to delete the user
+ * @param {*} endpoint Endpoint for GDPR request (https://support.appsflyer.com/hc/en-us/articles/360000811585#1-gdpr-request)
+ * @param {*} body default body which is same for all calls
+ * @param {*} identityType type of the identifier those are one of ios_advertising_id, android_advertising_id, appsflyer_id
+ * @param {*} identityValue value of identifier
+ * @returns
+ */
+const deleteUser = async (endpoint, body, identityType, identityValue) => {
+  body.subject_request_id = generateUUID();
+  body.submitted_time = new Date().toISOString();
+  body.subject_identities[0].identity_type = identityType;
+  body.subject_identities[0].identity_value = identityValue;
+  const response = await httpPOST(endpoint, body);
+  return response;
+};
+
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ * @param {*} userAttributes Array of objects with userId, emaail and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @returns
+ */
+const userDeletionHandler = async (userAttributes, config) => {
+  if (!Array.isArray(userAttributes)) {
+    throw new ErrorBuilder()
+      .setMessage("userAttributes is not an array")
+      .setStatus(400)
+      .build();
+  }
+  if (!config?.apiToken || !(config?.appleAppId || config?.androidAppId)) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "API Token and Apple ID or Android App Id are required fields user deletion"
+      )
+      .setStatus(400)
+      .build();
+  }
+  const body = {
+    subject_request_type: "erasure",
+    subject_identities: [{ identity_format: "raw" }]
+  };
+  if (config.statusCallbackUrls) {
+    const statusCallbackUrlsArray = config.statusCallbackUrls.split(",");
+    if (statusCallbackUrlsArray.length > 1) {
+      const filteredStatusCallbackUrlsArray = statusCallbackUrlsArray.filter(
+        statusCallbackUrl => {
+          const URLRegex = new RegExp(
+            "^(?:http(s)?:\\/\\/)?[\\w.-]+(?:\\.[\\w\\.-]+)+[\\w\\-\\._~:/?#[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$"
+          );
+          return statusCallbackUrl.match(URLRegex);
+        }
+      );
+      if (filteredStatusCallbackUrlsArray.length > 3) {
+        throw new ErrorBuilder()
+          .setMessage("you can send atmost 3 callBackUrls")
+          .setStatus(400)
+          .build();
+      }
+      body.status_callback_urls = filteredStatusCallbackUrlsArray;
+    }
+  }
+  const endpoint = `https://hq1.appsflyer.com/gdpr/opengdpr_requests?api_token=${config.apiToken}`;
+  for (let i = 0; i < userAttributes.length; i += 1) {
+    const userAttributeKeys = Object.keys(userAttributes[i]);
+
+    if (userAttributeKeys.includes("appsflyer_id")) {
+      body.property_id = config.androidAppId
+        ? config.androidAppId
+        : config.appleAppId;
```

Comment:
Add a check on `property_id` being not null

Reply:
This is actually already covered in the previous section. Where we are checking if anyone of `androidAppId` or `appleAppId` is present or not.